File: ZezeJava/ZezeJava/src/main/java/Zeze/Game/ProviderLoad.java
package Zeze.Game;

import Zeze.Arch.LoadConfig;
import Zeze.Arch.ProviderLoadBase;

public class ProviderLoad extends ProviderLoadBase {

	public final Online online;

	public ProviderLoad(Online online) {
		super(online.providerApp.zeze);
		this.online = online;
	}

	@Override
	public int getOnlineLocalCount() {
		return this.online.getLocalCount();
	}

	@Override
	public long getOnlineLoginTimes() {
		return this.online.getLoginTimes();
	}

	@Override
	public LoadConfig getLoadConfig() {
		return this.online.providerApp.distribute.loadConfig;
	}

	@Override
	public String getProviderIp() {
		return this.online.providerApp.directIp;
	}

	@Override
	public int getProviderPort() {
		return this.online.providerApp.directPort;
	}
}

File: Zeze/Game/ProviderLoad.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Zeze.Arch;
using Zeze.Builtin.Provider;
using Zeze.Net;
using Zeze.Serialize;

namespace Zeze.Game
{
	public class ProviderLoad : ProviderLoadBase
    {
        public Online Online { get; }

        public ProviderLoad(Online online)
            : base(online.ProviderApp.Zeze)
        {
            Online = online;
        }

        public override LoadConfig GetLoadConfig()
        {
            return Online.ProviderApp.Distribute.LoadConfig;
        }

        public override int GetOnlineLocalCount()
        {
            return Online.LocalCount;
        }

        public override long GetOnlineLoginTimes()
        {
            return Online.LoginTimes;
        }

        public override string GetProviderIp()
        {
            return Online.ProviderApp.DirectIp;
        }

        public override int GetProviderPort()
        {
            return Online.ProviderApp.DirectPort;
        }
    }
}
